docker/init_scripts/init

```diff
@@ -173,38 +174,54 @@ start_bin_scheduler() {
 		RQ_REDIS_SSL=${REDIS_SSL:-0} \
 		rqscheduler \
 		--path /backend \
-		--pid /tmp/scheduler.pid &
+		--pid /tmp/rq_scheduler.pid &
 }
 
-# function that runs our independent python scripts and creates corresponding PID files,
-start_python() {
-	SCRIPT="${1}"
-	info_log "Starting ${SCRIPT}"
-	python3 "${SCRIPT}.py" &
+# Commands to start RQ worker
+start_bin_rq_worker() {
+	info_log "Starting rq worker"
+
+	# Build Redis URL properly
+	local redis_url
+	if [[ -n ${REDIS_USERNAME-} && -n ${REDIS_PASSWORD-} ]]; then
+		redis_url="redis${REDIS_SSL:+s}://${REDIS_USERNAME}:${REDIS_PASSWORD}@${REDIS_HOST:-127.0.0.1}:${REDIS_PORT:-6379}/${REDIS_DB:-0}"
+	elif [[ -n ${REDIS_PASSWORD-} ]]; then
+		redis_url="redis${REDIS_SSL:+s}://:${REDIS_PASSWORD}@${REDIS_HOST:-127.0.0.1}:${REDIS_PORT:-6379}/${REDIS_DB:-0}"
+	else
+		redis_url="redis${REDIS_SSL:+s}://${REDIS_HOST:-127.0.0.1}:${REDIS_PORT:-6379}/${REDIS_DB:-0}"
+	fi
+
+	# Set PYTHONPATH so RQ can find the tasks module
+	PYTHONPATH="/backend:${PYTHONPATH-}" rq worker \
+		--path /backend \
+		--pid /tmp/rq_worker.pid \
+		--url "${redis_url}" \
+		high default low &
+}
+
+start_bin_watcher() {
+	info_log "Starting watcher"
+	watchmedo shell-command \
```

Comment:
Just in case, let's add the `watchmedo` extra for `watchdog` to our dependency definition:
https://github.com/gorakhargosh/watchdog/blob/d197487fb6b51aef05569bea0554f3272cb79039/setup.py#L64